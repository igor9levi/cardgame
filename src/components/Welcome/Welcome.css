.welcome-message {
    color: #fff;
}

button {
    margin: 20px;
}
