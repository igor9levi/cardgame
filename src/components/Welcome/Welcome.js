import React from 'react';
import PropTypes from 'prop-types';

export const Welcome = ({ startGame }) => (
  <div className="welcome-container">
    <h1 className="welcome-message">Hello! Please choose number of players</h1>
    <button
      id="btn1"
      type="button"
      onClick={() => {
        startGame(2);
      }}
    >
      Two Players
    </button>
    <button
      id="btn2"
      type="button"
      onClick={() => {
        startGame(3);
      }}
    >
      Three Players
    </button>
    <button
      id="btn3"
      type="button"
      onClick={() => {
        startGame(4);
      }}
    >
      Four Players
    </button>
  </div>
);

Welcome.propTypes = {
  startGame: PropTypes.func.isRequired,
};

export default Welcome;
