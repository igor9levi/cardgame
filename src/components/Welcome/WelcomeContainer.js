import { connect } from 'react-redux';
import { startGame } from '../App/appActions';
import WelcomeComponent from './Welcome';

export default connect(null, { startGame })(WelcomeComponent);
