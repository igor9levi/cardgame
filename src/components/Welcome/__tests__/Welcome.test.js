import React from 'react';
import { shallow } from 'enzyme';
import Welcome from '../Welcome';


describe('Test Welcome', () => {
  it('renders correctly', () => {
    const startGame = () => {};
    const wrapper = shallow(<Welcome startGame={startGame} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('sets 2 player game', () => {
    const startGame = jest.fn();
    const wrapper = shallow(<Welcome startGame={startGame} />);
    const buttons = wrapper.find('button');

    expect(buttons.length).toEqual(3);

    wrapper.find('#btn1').simulate('click');

    expect(startGame).toBeCalled();
    expect(startGame).toHaveBeenCalledTimes(1);
    expect(startGame).toBeCalledWith(2);

    wrapper.find('#btn2').simulate('click');

    expect(startGame).toBeCalled();
    expect(startGame).toHaveBeenCalledTimes(2);
    expect(startGame).toBeCalledWith(3);

    wrapper.find('#btn3').simulate('click');

    expect(startGame).toBeCalled();
    expect(startGame).toHaveBeenCalledTimes(3);
    expect(startGame).toBeCalledWith(4);
  });
});
