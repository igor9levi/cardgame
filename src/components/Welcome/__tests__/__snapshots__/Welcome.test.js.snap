// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Welcome renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    startGame={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="welcome-message"
>
          Hello! Please choose number of players
</h1>,
        <button
          id="btn1"
          onClick={[Function]}
          type="button"
>
          Two Players
</button>,
        <button
          id="btn2"
          onClick={[Function]}
          type="button"
>
          Three Players
</button>,
        <button
          id="btn3"
          onClick={[Function]}
          type="button"
>
          Four Players
</button>,
      ],
      "className": "welcome-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello! Please choose number of players",
          "className": "welcome-message",
        },
        "ref": null,
        "rendered": "Hello! Please choose number of players",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Two Players",
          "id": "btn1",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Two Players",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Three Players",
          "id": "btn2",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Three Players",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Four Players",
          "id": "btn3",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Four Players",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="welcome-message"
>
            Hello! Please choose number of players
</h1>,
          <button
            id="btn1"
            onClick={[Function]}
            type="button"
>
            Two Players
</button>,
          <button
            id="btn2"
            onClick={[Function]}
            type="button"
>
            Three Players
</button>,
          <button
            id="btn3"
            onClick={[Function]}
            type="button"
>
            Four Players
</button>,
        ],
        "className": "welcome-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello! Please choose number of players",
            "className": "welcome-message",
          },
          "ref": null,
          "rendered": "Hello! Please choose number of players",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Two Players",
            "id": "btn1",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Two Players",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Three Players",
            "id": "btn2",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Three Players",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Four Players",
            "id": "btn3",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Four Players",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
